src/transforms/regex_parser.rs

```diff
@@ -68,16 +73,56 @@ impl RegexParser {
             .as_ref()
             .unwrap_or(&event::log_schema().message_key());
 
-        let regex = Regex::new(&config.regex).context(super::InvalidRegex)?;
+        let regexes = match (&config.regex, &config.regexes.len()) {
+            (None, 0) => {
+                return Err(
+                    "At least one regular expression must be defined, but `regexes` is empty"
+                        .into(),
+                );
+            }
+            (None, _) => config.regexes.clone(),
+            (Some(regex), 0) => {
+                // Still using the old `regex` syntax.
+                // Printing a warning and wrapping input in a `vec`.
+                warn!(
+                    "Usage of `regex` is deprecated and will be removed in a future version. \
+                     Please upgrade your config to use `regexes` instead: \
+                     `regexes = ['{}']`",
```

Comment:
Shall we add a link to the [regex transform documentation](https://vector.dev/docs/reference/transforms/regex_parser/) here?

Reply:
Seems like a good idea!